### PR TITLE
[SDK]Fix default display name logic, use name as display name when not set

### PR DESCRIPTION
**Pull request by D-W-**
This pull request includes changes to improve the testing and functionality of the codebase. The most important changes include adding a new test method to assert the display name of a run, updating the default display name for runs in Promptflow, and modifying an assertion in a test method to include additional information.

Main changes:

* <a href="diffhunk://#diff-b63fdccdd1a3a293abe7799c0a231a7226614f726b6de1298f7e0233af26f128R156-R165">`src/promptflow/tests/sdk_cli_azure_test/e2etests/test_run_operations.py`</a>: Added a new test method `test_default_run_display_name` to assert the display name of a run.
* <a href="diffhunk://#diff-14f4d7a1d9b077c7b749230a92628782f990e1fc7cb741b20ff894f868327020L398-R398">`src/promptflow/promptflow/_sdk/entities/_run.py`</a>: Updated the default display name for runs in Promptflow to use the name of the run if the display name is not provided.
* <a href="diffhunk://#diff-94a59a05643476869fa3c6bc45466f1582944a935488075e2e63b6a6a196958fL510-R511">`src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py`</a>: Modified the assertion for the display name in the `test_run_display_name` method to include the name of the run and the string "print_env_var".

Other changes:

* <a href="diffhunk://#diff-8cebca1fd571c465e6f8c5d2ebb2ee85375dcd40884fab5ac02bed08caa3d689L27-L28">`examples/flows/standard/basic/flow.dag.yaml`</a>: Removed and then added the `python_requirements_txt` key under the `environment` section in the `flow.dag.yaml` file. <a href="diffhunk://#diff-8cebca1fd571c465e6f8c5d2ebb2ee85375dcd40884fab5ac02bed08caa3d689L27-L28">[1]</a> <a href="diffhunk://#diff-8cebca1fd571c465e6f8c5d2ebb2ee85375dcd40884fab5ac02bed08caa3d689R2-R3">[2]</a># Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: examples/flows/standard/basic/flow.dag.yaml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 $schema: https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json
+environment:
```
Comment (wangchao1230) [inline examples/flows/standard/basic/flow.dag.yaml:2]: add a changelog for this change

**Review by wangchao1230 — APPROVED** [review]

File: examples/flows/standard/basic/flow.dag.yaml (hunk 1 above)
Comment (wangchao1230) [reply examples/flows/standard/basic/flow.dag.yaml:2]: also add a changelog for flow as function feature

File: src/promptflow/promptflow/_sdk/entities/_run.py [hunk 2]
```diff
@@ -395,7 +395,7 @@ def _generate_run_name(self) -> str:
     def _get_default_display_name(self) -> str:
         display_name = self.display_name
         if not display_name:
-            display_name = self._get_flow_dir().name
+            display_name = self.name
```
Comment (0mza987) [inline src/promptflow/promptflow/_sdk/entities/_run.py:398]: can simple use `display_name = self.display_name or self.name`?

**Review by 0mza987 — APPROVED** [review]

File: src/promptflow/CHANGELOG.md [hunk 3]
```diff
@@ -8,10 +8,12 @@
 - Add support for image input and output in prompt flow.
 - [SDK/CLI] SDK/CLI will collect telemetry by default, user can use `pf config set telemetry.enabled=false` to opt out.
 - Add `raise_on_error` for stream run API, by default we raise for failed run.
+- Flow as function: consume a flow like a function with parameters mapped to flow inputs.
 
 ### Bugs Fixed
 
 - [SDK/CLI] Fix node test with dict node input will raise "Required input(s) missing".
+- [SDK/CLI] Will use run name as display name when display name not specified(used flow folder name before).
```
Comment (wangchao1230) [inline src/promptflow/CHANGELOG.md:16]: nit: should have a space here

specified(used

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/CHANGELOG.md (hunk 3 above)
Comment (D-W-) [reply src/promptflow/CHANGELOG.md:16]: Let me fix in next PR.

File: src/promptflow/promptflow/_sdk/entities/_run.py (hunk 2 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/entities/_run.py:398]: Let me fix in next PR.

**Pull request merged**